acceptance/test_state.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type testState struct {
+	contextDir string
+}
```

Comment:
Perhaps this should be an interface, then the `contextDir` would not be writable from this package? The interface could be something like:

```suggestion
func CreateTestState(contextDir string) (TestState, error) {
	ts := testStateImpl{contextDir}
	return ts, ts.mkdirs()
}

type TestState interface {
	SourceDir() string
	ArtifactsDir() string
	ResultsDir() string
	RestoredDir() string
	Teardown()
	ForMount(mountPath string) TestState
}
```

Reply:
Works for me, thanks!